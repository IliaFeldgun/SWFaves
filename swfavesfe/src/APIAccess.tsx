import { peopleURL, suggestionsURL } from "./config"
import { ICharacter } from "./interfaces/ICharacter";
import { IUserCharactersDTO } from "./interfaces/IUserCharactersDTO";

export async function getAllPeople() : Promise<ICharacter[]>
{
    let peopleResult = await fetch(peopleURL);
        
    return peopleResult.json();
}

export async function postFavorites(list: string[], userID: number)
{
    let parameters: IUserCharactersDTO = {
        _id: userID,
        characters: list
    }

    var response = await fetch(peopleURL, {
        method: 'Post',
        headers: {'Content-Type': 'application/json'},
        body: JSON.stringify(parameters)
    })

    return response.json();
}

export async function getSuggestedMovies(userID: number)
{
    let filmResult = await fetch(suggestionsURL + "/" + userID);

    return await filmResult.json();
}

export function getQueryString(param : string) : string
{
    const urlParams = new URLSearchParams(window.location.search);
    return urlParams.get(param)!;
}