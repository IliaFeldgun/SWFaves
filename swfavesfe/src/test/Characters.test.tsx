import React from 'react';
import ReactDOM, { render } from 'react-dom';
import App from '../routing'
import renderer from 'react-test-renderer'
import FavoriteCharacters, { CharacterList, SuggestButton } from '../Characters';
import { getAllPeople } from '../APIAccess';
import { BrowserRouter as Router } from 'react-router-dom';

it('renders a character list', () => {
    const characterList = renderer.create(<Router><CharacterList listSource={getAllPeople()} /></Router>).toJSON();
  
    expect(characterList).toMatchSnapshot();
  });

  it('renders a suggest button', () => {
    const suggestButton = renderer.create(<Router><SuggestButton FavoritesList={['Luke', 'Sky']} /></Router>).toJSON();
  
    expect(suggestButton).toMatchSnapshot();
  });

  it('renders Favourite Characters without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Router><FavoriteCharacters /></Router>, div);
    ReactDOM.unmountComponentAtNode(div);
  });