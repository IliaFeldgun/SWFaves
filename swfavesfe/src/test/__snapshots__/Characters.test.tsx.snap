// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a character list 1`] = `
<div
  id="characterlist"
>
  <h3>
    Search here:
  </h3>
  <input
    className="search-box"
    onChange={[Function]}
    type="text"
  />
  <div
    className="around-button"
  >
    <a
      href="/suggestedfilms?id=undefined"
      onClick={[Function]}
    >
      <button
        className="main-button"
        id="suggestbutton"
        onClick={[Function]}
      >
        Suggest me some movies!
      </button>
    </a>
  </div>
  <div
    id="characters"
  />
</div>
`;

exports[`renders a suggest button 1`] = `
<div
  className="around-button"
>
  <a
    href="/suggestedfilms?id=undefined"
    onClick={[Function]}
  >
    <button
      className="main-button"
      id="suggestbutton"
      onClick={[Function]}
    >
      Suggest me some movies!
    </button>
  </a>
</div>
`;
